perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/UsersManagerBlImpl.java

```diff
@@ -2064,43 +2061,38 @@ public void removeAllUserExtSources(PerunSession sess, User user) {
 	}
 
 	@Override
-	public String changePasswordRandom(PerunSession session, User user, String loginNamespace) throws PasswordOperationTimeoutException, LoginNotExistsException, PasswordChangeFailedException, InvalidLoginException, PasswordStrengthException {
-
-		char[] possibleCharacters =
-				"ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789!@#$%&*()-_=+;:,<.>/?"
-						.toCharArray();
-		int count = 12;
-
-		// FIXME - We will replace following logic once each login-namespace will implement
-		// FIXME   pwd-manager module and have server side checks
-		if (Objects.equals(loginNamespace, "vsup")) {
-			count = 14;
-			// removed O, l, specific only: +, -, *, /, .
-			possibleCharacters = "ABCDEFGHIJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz0123456789+-*/.".toCharArray();
-		}
-
-		String newRandomPassword = RandomStringUtils.random(count, 0, possibleCharacters.length - 1, false,
-				false, possibleCharacters, new SecureRandom());
+	public String changePasswordRandom(PerunSession session, User user, String namespace) throws PasswordOperationTimeoutException, LoginNotExistsException, PasswordChangeFailedException, InvalidLoginException, PasswordStrengthException {
 
+		// first check if user has login in specified namespace!
+		String userLogin;
 		try {
-			// FIXME - we want to re-generate password if it was considered weak
-			changePassword(session, user, loginNamespace, null, newRandomPassword, false);
-		} catch (PasswordDoesntMatchException | PasswordStrengthFailedException e) {
-			// should not happen when we are not using the old password
+			Attribute userLoginAttribute = getPerunBl().getAttributesManagerBl().getAttribute(session, user, AttributesManager.NS_USER_ATTR_DEF + ":login-namespace:" + namespace);
+			userLogin = (String) userLoginAttribute.getValue();
+		} catch (WrongAttributeAssignmentException | AttributeNotExistsException e) {
+			// should not happen since the changePassword method passed
```

Comment:
I can't see the changePassword method call.

Reply:
It was moved to the line 2087, since we now first check if user has login and then perform other logic.